crypto/client.go

```diff
@@ -15,12 +15,12 @@ type KeyDerivation interface {
 }
 
 func New(config *Config) (Client, error) {
-	switch config.Type {
+	switch t := config.Type; t {
 	default:
-		if config.Type != "" {
-			fmt.Println("Invalid encryption type, using default")
-		}
-		fallthrough
+		config.Type = DefaultConfigType
+		fmt.Printf("Invalid encryption type %q, using default %q\n", t, DefaultConfigType)
+		// This won't recurse indefinitely as we have provided a valid config type above
+		return New(config)
 	case ConfigTypeChaChaPoly:
 		// Remove other settings
 		// TODO(leon): This is shitty.
```

Comment:
Please remove these comments while you're in here

Reply:
It's still kind of shitty 😹. I'd prefer to not nil out those (unused) settings, however have yet to find a way to do so.

Reply:
I agree it's not elegant, but it does work! Perhaps you can reword it to something a bit nicer?

e.g.

```go
// TODO(leon): Remove other settings in a more elegant way
```

Reply:
I absolutely agree. Pushed another commit to clarify the `TODO` and removed any kind of swearing.